### ISSUE TITLE
Problem using HSL binaries with CasADi on Linux / CasADi Wiki improvement?

### ISSUE THREAD

**Issue by adbuerger** [casadi/casadi#1437]
Hello everyone,

I obtained the HSL binaries and followed the instructions [here](https://github.com/casadi/casadi/wiki/Obtaining-HSL) to get it working with IPOPT in CasADi (on Xubuntu 14.04). BLAS (``libblas3gf``, ``libblas-dev``), LAPACK (``liblapack3gf``, ``liblapack-dev``) and IPOPT (``coinor-libipopt-dev``) were installed via apt, ``/usr/lib/libhsl.so`` is a symbolic link to the ``libcoinhsl.so`` library and found by IPOPT.

Now, I run into the already mentioned problem of missing LAPACK(?) symbols. The answer about this in the [CasADi-FAQs](https://github.com/casadi/casadi/wiki/FAQ) is not really one, so my question is: did you get it working using the binaries on Linux?

If yes, it would be really great if you could point out to me how you achieved this, and we should then probably put this on the Wiki as well (if I can help you with this, please let me know!). And if not, maybe it would also be good to give a hint about this on the Wiki, so people will directly request the sources package.

Thanks you for your support!

**Comment by ghorn** [comment]
Can you please paste the error message? It's difficult to diagnose anything without it. Did you notice the lines in the wiki:

>   If you get errors about dtrmm_ not being found, try adding LIBS=-llapack to your configure command. The configure command I use is: ./configure --prefix=/home/ghorn/coinhsl/install_here_pls LIBS="-llapack" --with-blas="-L/usr/lib -lblas" CXXFLAGS="-g -O2 -fopenmp" FCFLAGS="-g -O2 -fopenmp"

**Comment by adbuerger** [comment]
Sorry about the missing error message, it's similar to the one you provided in a different issue:

    Exception of type: OPTION_INVALID in file "../../../../Ipopt/src/Algorithm/IpAlgBuilder.cpp" at line 345:
     Exception message: Selected linear solver MA57 not available.
    Tried to obtain MA57 from shared library "libhsl.so", but the following error occured:
    /usr/lib/libhsl.so: undefined symbol: dtrsm_

Note: I did not try to compile it! I directly obtained the binaries from the HSL page, not the sources, as it was described in the Wiki it would also be possible to directly use them. I now requested the sources from HSL and will try with the compiling instructions given in the Wiki. But for future readers, it might be best to remove the part about directly obtaining the binaries from the Wiki in case this doesn't work (or did it?), or add a part saying it won't work on Linux if it does for other systems.

**Comment by adbuerger** [comment]
Just for completion: your instructions for compiling work fine!

**Comment by b4be1** [comment]
Dear @ghorn, I think @adbuerger made a good point about updating the wiki. I had to wait a day till I get the binaries, then I've lost an hour because in the instruction for using binaries it is not mentioned, that one has to rename the .so file. After that I need to request the binaries and wait again.

**Comment by b4be1** [comment]
While following the instruction for installing HSL from binaries I encountered a problem when running step 6: "./configure.....". I did not have a fortran compiler installed and the LAPACK was not linking correctly. In case someone else encounters the same issue, here is what helped me:
1. Make sure the prerequisites from the casadi installation page on github are installed (meaning LAPACK and BLAS)
2. run the following: "ld -llapack --verbose" and make sure liblapack.so has been found somewhere. If not (you will see something like this http://linuxtoolkit.blogspot.de/2015/02/linker-returns-error-usrbinld-cannot.html), find where your LAPACK was installed. In my case it was in /user/lib/lapack and the file was called liblapack.so.3, so I had to create a symbolic link like this: "ln -s /usr/lib/liblapack.so.3 /usr/lib/liblapack.so"
2. install a fortran compiler (e.g. sudo apt-get install gfortran)

**Comment by adbuerger** [comment]
This indeed should be included in the wiki. I just helped someone installing the solver who had exactly the same problems, since he just installed CasADi by unzipping the compiled packages on a clean Ubuntu system he was missing blas, lapack and gfortran. In all cases, configuration of the build fails with a (for some parts misleading) message that gfortran does not work correctly. And he also did not get the precompiled HSL binaries working, which is what he tried before compiling the solvers. @jgillis would it be helpful for you if I updated the wiki accordingly, if you are short with time at the moment?
